contracts/CliptoExchange.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: AGPL-3.0 License
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./beacon/CloneableBeaconProxy.sol";
+import "./interfaces/ICliptoToken.sol";
+import "./CliptoExchangeStorage.sol";
+
+contract CliptoExchange is CliptoExchangeStorage, Initializable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+    uint256 private _feeNumer;
+    uint256 private _feeDenom;
+
+    modifier onlyOwner() {
+        require(owner == msg.sender, "not the owner");
+        _;
+    }
+
+    event OwnershipTransferred(address indexed oldOwner, address indexed newOwner);
+    event CreatorRegistered(address indexed creator, address indexed nft, string jsondata);
```

Comment:
Having user-modifiable jsondata here opens us up to impersonation attacks. We need validation somewhere in the pipeline here. If there's to be no backend with a DB, perhaps a centralized service which listens to twitter and flips a boolean somewhere on-chain?